src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerSource.java

```diff
@@ -224,6 +230,44 @@ public void onRunFinish(boolean succeeded, BatchSourceContext context) {
     }
   }
 
+  private List<?> transformArrayToList(ResultSet resultSet, String fieldName, Type arrayElementType) {
```

Comment:
there should also be unit tests for the conversion logic from a ResultSet into a StructuredRecord. Please extract out the conversion logic into a new class and add unit tests for it.

Reply:
Moved logic to ResultSetToRecordTransformer class, covered with tests in  ResultSetToRecordTransformerTest class